### PR TITLE
Patch github version of acvm with local copy

### PR DESCRIPTION
**Pull request by kevaundray**
Documentation on patch can be found here: https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section

This should mean that we do not need to supply the git commit for acvm in this crate. We supply the github version in each crate and then use patch to replace them with the local version. 

Removing ACVM from this repo would require one to remove the patch line and supply a different github link in each repository that needs it

### CODE REVIEW

**Comment by jfecher** [comment]
This looks good to me, I'll approve once clippy lints + conflicting files are fixed

**Comment by guipublic** [comment]
It turns out the change was finally in master, indeed after merging from master, we are left with one comment.

**Review by guipublic — APPROVED** [review]

**Pull request merged**
